docs/blueprints.rst

```diff
@@ -176,16 +176,24 @@ the `template_folder` parameter to the :class:`Blueprint` constructor::
 
     admin = Blueprint('admin', __name__, template_folder='templates')
 
-As for static files, the path can be absolute or relative to the blueprint
-resource folder.  The template folder is added to the searchpath of
-templates but with a lower priority than the actual application's template
-folder.  That way you can easily override templates that a blueprint
-provides in the actual application.
+For static files, the path can be absolute or relative to the blueprint
+resource folder.  
+
+The template folder is added to the search path of templates but with a lower 
+priority than the actual application's template folder. That way you can 
+easily override templates that a blueprint provides in the actual application. 
+This also means that you'll have to make sure that no blueprint template shares 
+the same relative path with any other blueprint or actual application template 
+else a name conflict will occur due to the way flask collects references to 
```

Comment:
The way this is currently worded might imply:
1. the "name conflict" will throw an exception
2. You must always make sure the names are unique, when in fact you might (rarely) wish to leverage this behavior. 

Maybe instead say something like:
> This also means that if you don't want a blueprint template to be accidentally overridden, make sure that no other blueprint or actual application template has the same relative path.